bot/exts/utilities/colour.py

```diff
@@ -0,0 +1,239 @@
+
+from bot.bot import Bot
+from bot.exts.core.extensions import invoke_help_command
+
+THUMBNAIL_SIZE = (80, 80)
+
+
+class Colour(commands.Cog):
+    """Cog for the Colour command."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        with open(pathlib.Path("bot/resources/utilities/ryanzec_colours.json")) as f:
+            self.colour_mapping = json.load(f)
+            del self.colour_mapping['_']  # Delete source credit entry
+
+    async def send_colour_response(self, ctx: commands.Context, rgb: tuple[int, int, int]) -> None:
+        """Create and send embed from user given colour information."""
+        name = self._rgb_to_name(rgb)
+
+        try:
+            colour_or_color = ctx.invoked_parents[0]
+        except IndexError:
+            colour_or_color = "colour"
+
+        colour_mode = ctx.invoked_with
+        if colour_mode == "random":
+            input_colour = "random"
+        elif colour_mode in ("colour", "color"):
+            input_colour = rgb
+        elif colour_mode == "name":
+            input_colour = ctx.kwargs["user_colour_name"]
+        else:
+            input_colour = tuple(ctx.args[2:])
+
+        if colour_mode not in ("name", "hex", "random", "color", "colour"):
+            colour_mode = colour_mode.upper()
+        else:
+            colour_mode = colour_mode.title()
+
+        colour_embed = discord.Embed(
+            title=colour_or_color.title(),
+            description=f"{colour_mode} information for `{name or input_colour}`.",
+            colour=discord.Color.from_rgb(*rgb)
+        )
+        colour_conversions = self.get_colour_conversions(rgb)
+        for colour_space, value in colour_conversions.items():
+            colour_embed.add_field(
+                name=colour_space,
+                value=f"`{value}`",
+                inline=True
+            )
+
+        thumbnail = Image.new("RGB", THUMBNAIL_SIZE, color=rgb)
+        buffer = BytesIO()
+        thumbnail.save(buffer, "PNG")
+        buffer.seek(0)
+        thumbnail_file = discord.File(buffer, filename="colour.png")
+
+        colour_embed.set_thumbnail(url="attachment://colour.png")
+
+        await ctx.send(file=thumbnail_file, embed=colour_embed)
+
+    @commands.group(aliases=("color",), invoke_without_command=True)
+    async def colour(self, ctx: commands.Context, *, extra: str) -> None:
+        """User initiated command to create an embed that displays colour information."""
+        if ctx.invoked_subcommand is None:
+            try:
+                extra_colour = ImageColor.getrgb(extra)
+                await self.send_colour_response(ctx, extra_colour)
+            except ValueError:
+                await invoke_help_command(ctx)
+
+    @colour.command()
+    async def rgb(self, ctx: commands.Context, red: int, green: int, blue: int) -> None:
+        """Command to create an embed from an RGB input."""
+        if any(c not in range(0, 256) for c in (red, green, blue)):
```

Comment:
This has been repeated in a number of places.

```suggestion
        if any(c not in range(256) for c in (red, green, blue)):
```

Reply:
Resolved with [93e6e96](https://github.com/python-discord/sir-lancebot/pull/842/commits/93e6e96c89e621db6e463f4d5716533fd85cd96f)